### ISSUE TITLE
[Caffe2] resize5d stride/dim issue

### ISSUE THREAD

**Issue by NattapongSiri** [pytorch/pytorch#16138]
## 🐛 Bug

THTensor resize4d and resize5d functions cause a corrupted tensor.
The mentioned function defined here: https://github.com/pytorch/pytorch/blob/ded4ff87af6e4cd0d22ef091f8ba7d49a4a49d1c/aten/src/TH/generic/THTensor.h#L75-L76
The return tensor will cause a crash when I call to nDimension or stride functions.
The dim and stride functions are defined here: https://github.com/pytorch/pytorch/blob/ded4ff87af6e4cd0d22ef091f8ba7d49a4a49d1c/aten/src/TH/generic/THTensor.h#L27
https://github.com/pytorch/pytorch/blob/ded4ff87af6e4cd0d22ef091f8ba7d49a4a49d1c/aten/src/TH/generic/THTensor.h#L31
All other resize work fine.
## To Reproduce

1. THFloatStorage_newWithSize(10)
1. THFloatTensor_newWithStorage1d(PREV_STORAGE, 0, 10, 1)
1. THFloatTensor_resize5d(PREV_TENSOR, 1, 2, 1, 2, 1)
1. THFloatTensor_nDimension(PREV_TENSOR)
crash...

## Expected behavior
Return dimension of tensor.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.0
 - OS (e.g., Linux): Windows 10.0.17134.523
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6.7
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information:

One observed behavior for resize4d, if source tensor is created as 4d, it crashed. When source tensor has different dimension, e.g. 3d, it fine.
However, resize5d always return corrupted tensor.

## Additional context
I link to caffe2 lib by copy following files into project path:
c10.dll
caffe2.dll
caffe2.lib
libiomp5md.dll

I tried to create Torch Tensor inside Rust by using Rust FFI binding to C language. This is done to reduce cost of constructing tensor object inside Python as my experiment data source, if loaded entirely in Python consume too much memory, i.e. 8 bytes per an element. In C/Rust, I can use single byte per an element. Loading data then GC it make my training time slow down from few hours/epoch to 5 days per epoch.
So far, all other resize* work fine except these two.

Side question: does this resize function copy data ? It seem like all resized tensor always contiguous.
